columnar/src/backend/columnar/columnar.c

```diff
@@ -49,6 +49,7 @@ bool columnar_enable_vectorization = true;
 bool columnar_enable_dml = true;
 bool columnar_enable_page_cache = true;
 int columnar_page_cache_size = 200U;
+bool columnar_index_scan = false;
```

Comment:
why is this disabled by default?

Reply:
My opinion is that, because it changes how planning cost is done, it should be used only when user try this optimization and sees does it help for specific query.

Reply:
I agree with this opinion.  we can re-evaluate later, but given that we're mucking with costing, it should be a slow and deliberate rollout.  mucking with costing is one of the scariest things we can do (other than cause corruption).

Reply:
Why would you categorize it as "scary"? I know there is risk that a query may run slower (or indefinitely)--what other risk is there?

Reply:
I categorize it as "scary" because we _can have_ altered the costing overall enough that we end up with a slowdown across a majority of queries.  I know that we benchmark against 2 benchmarks (clickbench and "warehouse"), but those are not necessarily real-world tests.  I am a little risk-averse, and would feel more comfortable if we had more feedback and real-world examples before enabling it by default.

costing changes, as we've shown in the past, can end up with a general slowdown, so I personally prefer to be extra-careful when dealing with costing.  a GUC fulfills that, as well as giving us the ability to see real-world impact before enabling it by default.

if we don't see general slowdowns after some set time period, I'd be in favor of enabling it by default.